beancount_reds_importers/libreader/reader.py

```diff
@@ -31,23 +31,21 @@ def file_name(self, file):
         return '{}'.format(ntpath.basename(file.name))
 
     def file_account(self, _):
-        # Ugly hack to handle:
+        # Ugly hack to handle an interaction with smart_importer. See:
         # https://github.com/redstreet/beancount_reds_importers/issues/41
         # https://github.com/beancount/smart_importer/issues/122
         # https://github.com/beancount/smart_importer/issues/30
         import inspect
         curframe = inspect.currentframe()
         calframe = inspect.getouterframes(curframe, 2)
-
-        # smart_importer call
         if any('predictor' in i.filename for i in calframe):
             if 'smart_importer_hack' in self.config:
                 return self.config['smart_importer_hack']
 
-        # bean-file call
+        # Otherwise handle a typical bean-file call
         if 'filing_account' in self.config:
             return self.config['filing_account']
-        return self.config['main_account'].replace(':{ticker}', '').replace(':{currency}', '')
+        return self.config['main_account']
```

Comment:
The change to investments.py is to set filing_account instead, so that the replacement is localized to investments. The regression tests show the filename is correct.

Reply:
Ah, this is dead code then. Would you mind removing it and simply returning `self.config[filing_account]`?